plugins/ltac/tactic_option.ml

```diff
@@ -8,40 +8,65 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
+open Goptions
 open Libobject
-open Pp
+
+let tac_option_locality = Attributes.option_locality
+
+let warn_default_locality =
+  CWarnings.create ~name:"deprecated-tacopt-without-locality" ~category:"deprecated" Pp.(fun () ->
+      strbrk
+        "The default locality for this command outside sections is \
+         currently like the combination of \"global\", \"export\" and \
+         re-exporting for every surrounding module. It will change to \
+         \"export\" in a future release.")
```

Comment:
Given the change to the meaning of `global`, adding the deprecation warning now is problematic because there is no backward-compatible way of fixing it. It seems that a better strategy would be to release this PR without the warning and to add the warning in the next release. BTW, I believe that most other warnings about the default locality do not mention as explicitly that the plan is for the new locality to be `export`, giving us some freedom to discuss making the default locality `local` instead (see part of the discussion at #13394).

Reply:
>Given the change to the meaning of global, adding the deprecation warning now is problematic because there is no backward-compatible way of fixing it. It seems that a better strategy would be to release this PR without the warning and to add the warning in the next release.

I don't understand what the point of delaying is.

Reply:
Users usually do not like getting warnings that they cannot fix in backward-compatible ways. But maybe this is not such a big deal as users can choose between dropping compatibility with previous Coq releases, or disabling the warning until they do.